nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/NucleusTicketQuery.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.api.query;
+
+import com.google.common.collect.ImmutableMap;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.util.ResettableBuilder;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Map;
+
+public interface NucleusTicketQuery {
```

Comment:
This is not at a high enough level. Plugins should never EVER care about how the tickets are stored. Plugins consuming the API should only care about the result, not the means - and you're restricting the means here. Say I later come back to implement a MongoDB backend on this. MongoDB does not use SQL, so this method is useless. Same if I go for flat file.

The idea of the API is that is simply gives you the results. It doesn't tell you the how it's implemented or the why, it just provides the what.

Such a query interface (which could be useful in constructing complex queries) should potentially hold a collection of filter interfaces. These filter interfaces  would also not expose SQL details and the like, but an interface will represent a condition, so it would represent a filter where the creation date is after a week ago. The API doesn't say how it will achieve that - the _implementation_ will, and that is completely hidden from the API.

So, absolutely _nothing_ to do with SQL should appear in the API.